### PR TITLE
feat: add sample for use_sim

### PR DESCRIPTION
**Pull request by takeshi-iwanari**
Signed-off-by: takeshi.iwanari <takeshi.iwanari@tier4.jp>

I modified sample code, so that it can be used to demonstrate use_sim in CARET_doc. This modification should make no effect on current implementation. My only concern is this modification make the current code little bit complicated. Demo code should be simple though..

Reference: https://tier4.github.io/CARET_doc/pr-84/recording/sim_time/#sample-project-to-use-sim_time  (WIP, and this branch will disappear in the future)

### CODE REVIEW

File: caret_demos/src/end_to_end_sample.cpp [hunk 1]
```diff
@@ -139,6 +139,14 @@ class SensorDummy : public rclcpp::Node
   SensorDummy(std::string node_name, std::string topic_name, int period_ms)
   : Node(node_name)
   {
+    this->declare_parameter<bool>("use_rosbag", false);
+    bool use_rosbag;
```
Comment (hsgwa) [inline caret_demos/src/end_to_end_sample.cpp]: here, a warning occurs.

```
/home/hasegawa/ros2_ws/src/CARET_demos/caret_demos/src/end_to_end_sample.cpp:146:5: warning: ‘use_rosbag’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  146 |     if (use_rosbag) {
```

**Review by hsgwa — CHANGES_REQUESTED** [review]
> My only concern is this modification make the current code little bit complicated. 

Maybe, using clock_publisher might keep sample code simple.

This is just my sample node which publishes /clock topic.
https://github.com/hsgwa/clock_publisher

It's may good to implement clock_publisher to CARET_demos and omit rosbag record&play procedure.

File: caret_demos/src/end_to_end_sample.cpp (hunk 1 above)
Comment (takeshi-iwanari) [reply caret_demos/src/end_to_end_sample.cpp]: Thank you for pointing it out. The warning doesn't appear in my environment but the new commit should fix it.

Also, thank you for your idea using clock_publisher. It looks nice to have it but I think most users will perform rosbag record/play when using use_sim. So, I'd like to keep this procedure as of current implementation.

**Review by hsgwa — APPROVED** [review]
LGTM!

**Pull request merged**